src/main/java/de/tum/in/www1/artemis/web/rest/dto/learningpath/NgxLearningPathDTO.java

```diff
@@ -0,0 +1,114 @@
+package de.tum.in.www1.artemis.web.rest.dto.learningpath;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+/**
+ * Represents simplified learning path optimized for Ngx representation
+ */
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public record NgxLearningPathDTO(Set<Node> nodes, Set<Edge> edges, Set<Cluster> clusters) {
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(nodes, edges, clusters);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (!(obj instanceof NgxLearningPathDTO other)) {
+            return false;
+        }
+        return nodes.equals(other.nodes) && edges.equals(other.edges) && clusters.equals(other.clusters);
+    }
+
+    @Override
+    public String toString() {
+        return "NgxLearningPathDTO{nodes=" + nodes + ", edges=" + edges + ", clusters=" + clusters + "}";
+    }
```

Comment:
Aren't those automatically generated in records?
The toString might be valid depending on your needs for the representation